File: EClassBook.Model/Course.cs
namespace EClassBook.Model
{
    using Common.Models;
    using System.Collections.Generic;

    public class Course : BaseModel
    {
        private ICollection<Grade> grades;

        public Course()
        {
            this.grades = new HashSet<Grade>();
        }

        public string Name { get; set; }

        public int UserId { get; set; }

        public virtual User User { get; set; }

        public virtual ICollection<Grade> Grades { get; set; }
    }
}

File: EClassBook.API/wwwroot/app/core/models/course.ts
export class Course {
    Id: number;
    Name: string;
    UserId: number;
    TeacherName: string;

    constructor(name: string, userId: number, id?: number, teacherName?: string) {
        this.Id = id;
        this.Name = name;
        this.UserId = userId;
        this.TeacherName = teacherName;
    }
}